bluemira/base/design.py

```diff
@@ -67,9 +67,16 @@ def __init__(
         self._build_config = copy.deepcopy(build_config)
         self._extract_build_config(params)
         self._validate_params(params)
-        self._params = Configuration.from_template(self._required_params)
+        self._params = self.configuration.from_template(self._required_params)
         self._params.update_kw_parameters(params)
 
+    @abc.abstractproperty
+    def configuration(self) -> type(Configuration):
+        """
+        Configuration
```

Comment:
I realise this may only be here temporarily, but it may help in refactoring later if we spelled the purpose of this configuration out a little. It's not immediately obvious to me.

Is it all parameters for the `Design`? Or the local? Or global?

Reply:
Also, I think this should be `Type[Configuration]`